Sources/Image/GIFAnimatedImage.swift

```diff
@@ -116,4 +116,13 @@ class GIFAnimatedImage {
         guard let frameDuration = duration else { return defaultFrameDuration }
         return frameDuration.doubleValue > 0.011 ? frameDuration.doubleValue : defaultFrameDuration
     }
+
+    // Calculates frame duration at a specific index for a gif from an `imageSource`.
+    static func getFrameDuration(from imageSource: CGImageSource, at index: Int) -> TimeInterval {
```

Comment:
Could you also refactor the code in `init` to use this as well? Now they are seemly a bit duplicated.

Reply:
sure